tests/queries/0_stateless/02354_numeric_literals_with_underscores.sql

```diff
@@ -0,0 +1,6 @@
+select 1_000;
+select 1.00_00_01;
+select 1.000_0001e2;
+select 0x12_34_56_78;
+select 0x12_34_56_78p1;
+select 0b0010_0100_0111;
```

Comment:
Let's also add negative testcases
Maybe you'll find more corner cases

```suggestion
select 0b0010_0100_0111;

select _1000; -- { serverError UNKNOWN_IDENTIFIER }
select _1000 FROM (SELECT 1 AS _1000) FORMAT Null;

select -_1; -- { serverError UNKNOWN_IDENTIFIER }
select -_1 FROM (SELECT -1 AS _1) FORMAT Null;

select +_1; -- { clientError SYNTAX_ERROR }

select 1____0; -- { clientError SYNTAX_ERROR }
select 10_; -- { clientError SYNTAX_ERROR }
select 1_e5; -- { clientError SYNTAX_ERROR }
select 1e_5; -- { clientError SYNTAX_ERROR }
select 1e_; -- { clientError SYNTAX_ERROR }
select 0_x2; -- { clientError SYNTAX_ERROR }
select 0x_; -- { clientError SYNTAX_ERROR }

```